### DISCUSSION TITLE
Quick erase (clean partition table)

### DISCUSSION THREAD

**Discussion by slym** [PartialVolume/shredos.x86_64#288] [Ideas]
Hello there, 

I tried ShredOS and it is great however I was looking for a quick erase option for testing purposes.

Could you add an "unsecure and quick" erase option that would only clean/reset partition tables ?

My use-case is quite simple : for testing purposes I have to flash the same image again and again but want to start without partitions to simulate a fresh install on new hardware.

No security is needed as there is no confidential data on disks.

I know I could do that with command lines on other Live CD but was looking for a quicker way to do that with a menu.

Thank you in advance.

**Comment by PartialVolume** [comment]
That could be added as a method, however I would want to hide it from the menu unless you specifically placed a nwipe option on the command line such as --enable-insecure-methods and if you did provide that command line option the method would then show up in the gui.

My reasoning being that I wouldn't want somebody that wasn't sure of what they were doing to accidentally use this insecure method thinking they had wiped their drive.

It would wipe out the first ~32 blocks which would wipe out the MBR and partition table and would take a fraction of a second to complete the wipe. The PDF would provide all the warnings that the drive might appear empty from the OS point of view but with the right forensic recovery tools all the data can be recovered.

It's not high on my priority list so if anybody else that's familiar with nwipe's code fancies adding this as a method let me know otherwise it will have to wait until after I've completed the extra host specific PDF and secure erase code. Unfortunately I can't provide any timeline for when that might be.

https://github.com/users/PartialVolume/projects/1/views/1?pane=issue&itemId=79986959

**Comment by slym** [comment]
Great ! thank you for considering my request for your roadmap. I'm not in a hurry so no problem to keep it in low priority.

However I'm not sure that cleaning only the first 32 blocks will be enough. In my first test I used one of the standard erasure method and stopped it after a few seconds but some partitions were still visible (in tools such as gparted). I retried and waited for a few minutes instead and then it was enough and there were no trace left.

Also, in my opinion the ShredOS menu is sufficiently clear so that this new option could be displayed. Its name could contain a "UNSECURED" string to avoid any issue + you also already display the security level in the description. It's more than enough for users to understand that this option is not a secured method ;-)

**Reply by PartialVolume** [reply]
>However I'm not sure that cleaning only the first 32 blocks will be enough. In my first test I used one of the standard erasure method and stopped it after a few seconds but some partitions were still visible (in tools such as gparted). I retried and waited for a few minutes instead and then it was enough and there were no trace left.

Nwipe uses cached I/O rather than direct I/O so CPU memory size has a lot to do with how long you wait. After you see the first sync, 100,000 blocks have been written to disc but before that first sync completes, anything from 0 to 100,000 block may have been written.
